addons/popochiu/editor/main_dock/popochiu_row/object_row/room_object_row/popochiu_room_object_row.gd

```diff
@@ -0,0 +1,131 @@
+@tool
+extends "res://addons/popochiu/editor/main_dock/popochiu_row/object_row/popochiu_object_row.gd"
+
+enum RoomObjectOptions {
+	DELETE = MenuOptions.DELETE,
+	ADD_TO_CORE = Options.ADD_TO_CORE,
+	CREATE_STATE_SCRIPT = Options.CREATE_STATE_SCRIPT,
+	CREATE_SCRIPT,
+}
+
+const PROP_TEMPLATE = "res://addons/popochiu/engine/templates/prop_template.gd"
+
+var node_path := ""
+
+
+#region Godot ######################################################################################
+func _ready() -> void:
+	super()
+	
+	if ".gd" in path or ".tscn" in path:
+		menu_popup.remove_item(menu_popup.get_item_index(RoomObjectOptions.CREATE_SCRIPT))
+	
+	if not FileAccess.file_exists(path.replace(".tscn", ".gd")):
+		btn_script.hide()
+	
+	btn_state.hide()
+	btn_state_script.hide()
+
+
+#endregion
+
+#region Virtual ####################################################################################
+func _get_location() -> String:
+	# Structure of path: "res://game/rooms/room_name/props/prop_name/"
+	# path splitted: [res:, popochiu, rooms, room_name, props, prop_name]
+	return "Room%s" % (path.split("/", false)[3]).to_pascal_case()
+
+
+#endregion
+
+#region Private ####################################################################################
+func _get_menu_cfg() -> Array:
+	return [
+		{
+			id = RoomObjectOptions.CREATE_SCRIPT,
+			icon = get_theme_icon("ScriptCreate", "EditorIcons"),
+			label = "Create script",
+		},
+	] + super()
+
+
+func _menu_item_pressed(id: int) -> void:
+	match id:
+		RoomObjectOptions.CREATE_SCRIPT:
+			var prop_template := load(PROP_TEMPLATE)
```

Comment:
Props are the only ones that can not have an attached script?
Is it worth to make without this assumption and give a script to every prop, just making some of them non-clickable? 

This will spare us from maintaining all the code to add or remove scripts.
Non-clickable props may still have scripts for time-bound or signal-based automation.

Reply:
Yes. This is an old code no longer needed. I forgot to remove it. Now all props have a script by default.